files/usr/share/jeos-firstboot/raspberrywifi

```diff
@@ -122,23 +120,40 @@ WIRELESS_ESSID='$wlan_network'
 WIRELESS_MODE='Managed'
 EOF
 
-	if [ $wlan_auth_mode = "WPA-PSK" ]; then
-		echo "WIRELESS_WPA_PSK='$wlan_password'" >> $config_file
-	fi
+		if [ $wlan_auth_mode = "WPA-PSK" ]; then
+			echo "WIRELESS_WPA_PSK='$wlan_password'" >> $config_file
+		fi
 
-	if [ $wlan_auth_mode = "WPA-EAP" ]; then
-		echo "WIRELESS_WPA_IDENTITY='$wlan_username'" >> $config_file
-		echo "WIRELESS_WPA_PASSWORD='$wlan_password'" >> $config_file
-		echo "WIRELESS_EAP_AUTH='mschapv2'" >> $config_file
-	fi
+		if [ $wlan_auth_mode = "WPA-EAP" ]; then
+			echo "WIRELESS_WPA_IDENTITY='$wlan_username'" >> $config_file
+			echo "WIRELESS_WPA_PASSWORD='$wlan_password'" >> $config_file
+			echo "WIRELESS_EAP_AUTH='mschapv2'" >> $config_file
+		fi
 
-	run mv -f $config_file /etc/sysconfig/network/ifcfg-$wlan_device
+		run mv -f $config_file /etc/sysconfig/network/ifcfg-$wlan_device
+		d --infobox $"Connecting to wireless network ..." 3 38 || true
+
+		run ifdown $wlan_device &>/dev/null || true
+		# Disabling exit on error to capture exit code
+		set +e
+		ifup $wlan_device &>/dev/null
+		ret_val=$?
+		set -e
+		if [ $ret_val -ne 0 ]; then
+			if dialog --yesno $"Connection failed, do you wish to retry?" 0 0; then
+				continue
+			else
+				# Otherwise, return 0 to continue main execution
+				return 0
```

Comment:
Please use `break` and `return 0` consistently

Reply:
I do not understand what is inconsistent here, can you please clarify?

Reply:
You use `break` below to achieve the same.